core/storage/beam_job/gae_models.py

```diff
@@ -0,0 +1,165 @@
+# limitations under the License.
+
+"""Models for managing Apache Beam jobs."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+datastore_services = models.Registry.import_datastore_services()
+
+# NOTE: The following values are constants from an enum defined by Google Cloud
+# Dataflow, and are thus outside of our control:
+# https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#jobstate
+
+# The job is currently running.
+BEAM_JOB_STATE_RUNNING = 'RUNNING'
+# The job has been created but is not yet running. Jobs that are pending may
+# only transition to RUNNING, or FAILED.
+BEAM_JOB_STATE_PENDING = 'PENDING'
+# The job has not yet started to run.
+BEAM_JOB_STATE_STOPPED = 'STOPPED'
+# The job has has been explicitly cancelled and is in the process of stopping.
+# Jobs that are cancelling may only transition to CANCELLED or FAILED.
+BEAM_JOB_STATE_CANCELLING = 'CANCELLING'
+# The job has has been explicitly cancelled. This is a terminal job state. This
+# state may only be set via a Cloud Dataflow jobs.update call, and only if the
+# job has not yet reached another terminal state.
+BEAM_JOB_STATE_CANCELLED = 'CANCELLED'
+# The job is in the process of draining. A draining job has stopped pulling from
+# its input sources and is processing any data that remains in-flight. This
+# state may be set via a Cloud Dataflow jobs.update call, but only as a
+# transition from RUNNING. Jobs that are draining may only transition
+# to DRAINED, CANCELLED, or FAILED.
+BEAM_JOB_STATE_DRAINING = 'DRAINING'
+# The job has been drained. A drained job terminated by stopping pulling from
+# its input sources and processing any data that remained in-flight when
+# draining was requested. This state is a terminal state, may only be set by the
+# Cloud Dataflow service, and only as a transition from DRAINING.
+BEAM_JOB_STATE_DRAINED = 'DRAINED'
+# The job was successfully updated, meaning that this job was stopped and
+# another job was started, inheriting state from this one. This is a terminal
+# job state. This state may only be set by the Cloud Dataflow service, and only
+# as a transition from RUNNING.
+BEAM_JOB_STATE_UPDATED = 'UPDATED'
+# The job has successfully completed. This is a terminal job state. This state
+# may be set by the Cloud Dataflow service, as a transition from
+# RUNNING. It may also be set via a Cloud Dataflow jobs.update call,
+# if the job has not yet reached a terminal state.
+BEAM_JOB_STATE_DONE = 'DONE'
+# The job has has failed. This is a terminal job state. This state may only be
+# set by the Cloud Dataflow service, and only as a transition from RUNNING.
+BEAM_JOB_STATE_FAILED = 'FAILED'
+# The job's run state isn't specified.
+BEAM_JOB_STATE_UNKNOWN = 'UNKNOWN'
```

Comment:
Should this also be an enum with all the possible values?

Reply:
Done.